src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,157 @@
+        Optional<BSPKitStatus> result = Optional.empty();
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId())) {
+                String message = "Kit of exited participant " + maybeBspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+                notificationUtil.sentNotification(maybeBspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.EXITED));
+            }
+            else if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.DEACTIVATED));
+            }
+        }
+        return result;
+    }
+
+    public boolean canReceiveKit(@NonNull String kitLabel) {
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return false;
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId()) || StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                logger.info("Kit can not be received.");
+                return false;
+            }
+        }
+        return true;
+
+    }
+
+    public Optional<BSPKitInfo> receiveBSPKit(String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Trying to receive kit " + kitLabel);
+        return TransactionWrapper.inTransaction(conn -> {
+            Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+             BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDdpParticipantId())) {
+                boolean firstTimeReceived = KitUtil.setKitReceived(conn, kitLabel);
+                DDPInstance ddpInstance = DDPInstance.getDDPInstance(maybeBspKitQueryResult.getInstanceName());
+                InstanceSettings instanceSettings = new InstanceSettings();
+                InstanceSettingsDto instanceSettingsDto = instanceSettings.getInstanceSettings(maybeBspKitQueryResult.getInstanceName());
+                Value received = instanceSettingsDto
+                        .getKitBehaviorChange()
+                        .map(kitBehavior -> kitBehavior.stream().filter(o -> o.getName().equals(InstanceSettings.INSTANCE_SETTING_RECEIVED)).findFirst().orElse(null))
+                        .orElse(null);
```

Comment:
instead of using `orElse(null)` here, I would use `ifPresentOrElse` method which would bring much better structure than `if (received != null) {...} else {...}`.